tests/unit/query_v2_query_plan_match_filter_return.cpp

```diff
@@ -0,0 +1,2062 @@
+  }
+  {
+    // Upper bound isn't property value
+    auto scan_index = MakeScanAllByLabelPropertyRange(storage, symbol_table, "n", label1, prop, "prop", std::nullopt,
+                                                      Bound{ident_m, Bound::Type::INCLUSIVE}, scan_all.op_);
+    auto context = MakeContext(storage, symbol_table, &dba);
+    EXPECT_THROW(PullAll(*scan_index.op_, &context), QueryRuntimeException);
+  }
+  {
+    // Both bounds aren't property value
+    auto scan_index = MakeScanAllByLabelPropertyRange(storage, symbol_table, "n", label1, prop, "prop",
+                                                      Bound{ident_m, Bound::Type::INCLUSIVE},
+                                                      Bound{ident_m, Bound::Type::INCLUSIVE}, scan_all.op_);
+    auto context = MakeContext(storage, symbol_table, &dba);
+    EXPECT_THROW(PullAll(*scan_index.op_, &context), QueryRuntimeException);
+  }
+}
+
+TEST_F(QueryPlanMatchFilterTest, ScanAllByLabelPropertyEqualNull) {
+  // Add 2 vertices with the same label, but one has a property value while
+  // the other does not. Checking if the value is equal to null, should
+  // yield no results.
+  auto label1 = db.NameToLabel("label1");
+  auto prop = db.NameToProperty("prop");
+  {
+    auto storage_dba = db.Access();
+    DbAccessor dba(&storage_dba);
+    auto vertex = *dba.InsertVertexAndValidate(label, {}, {{property, storage::PropertyValue(1)}});
+    ASSERT_TRUE(vertex.AddLabel(label1).HasValue());
+    auto vertex_with_prop = *dba.InsertVertexAndValidate(label, {}, {{property, storage::PropertyValue(1)}});
+    ASSERT_TRUE(vertex_with_prop.AddLabel(label1).HasValue());
+    ASSERT_TRUE(vertex_with_prop.SetProperty(prop, storage::PropertyValue(42)).HasValue());
+    ASSERT_FALSE(dba.Commit().HasError());
+  }
+  db.CreateIndex(label1, prop);
+
+  auto storage_dba = db.Access();
+  DbAccessor dba(&storage_dba);
+  EXPECT_EQ(2, CountIterable(dba.Vertices(storage::View::OLD)));
+  // MATCH (n :label1 {prop: 42})
+  AstStorage storage;
+  SymbolTable symbol_table;
+  auto scan_all =
+      MakeScanAllByLabelPropertyValue(storage, symbol_table, "n", label1, prop, "prop", LITERAL(TypedValue()));
+  // RETURN n
+  auto output = NEXPR("n", IDENT("n")->MapTo(scan_all.sym_))->MapTo(symbol_table.CreateSymbol("n", true));
+  auto produce = MakeProduce(scan_all.op_, output);
+  auto context = MakeContext(storage, symbol_table, &dba);
+  auto results = CollectProduce(*produce, &context);
+  EXPECT_EQ(results.size(), 0);
+}
+
+TEST_F(QueryPlanMatchFilterTest, ScanAllByLabelPropertyRangeNull) {
+  // Add 2 vertices with the same label, but one has a property value while
+  // the other does not. Checking if the value is between nulls, should
+  // yield no results.
+  auto label1 = db.NameToLabel("label1");
+  auto prop = db.NameToProperty("prop");
+  {
+    auto storage_dba = db.Access();
+    DbAccessor dba(&storage_dba);
+    auto vertex = dba.InsertVertex();
```

Comment:
Please replace this with `InsertVertexAndValidate`.

Reply:
I've missed that sorry